modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/result/AbstractReturnDelegate.java

```diff
@@ -293,13 +302,36 @@ protected MediaType resolveMediaType(Object value, MediaType contextMimeType, Ch
     return contextMimeType.withCharset(contextEncoding);
   }
 
+  private Runnable incrementActiveComponent(ExecutionContextAdapter executionContext) {
```

Comment:
this is inefficient performance wise. You should check if the config is present or not before hand. If it's present, contruct a runnable using the config directly, avoiding the Optional#map call. If it's not present, do not create an empty `Runnable`,  simply pass `null` and have the stream check for that.